app/controllers/users_controller.rb

```diff
@@ -113,6 +113,18 @@ def extlogout
     render :extlogout, :layout => 'login'
   end
 
+  def test_mail
+    user = find_resource(:edit_users)
+    begin
+      MailNotification[:tester].deliver(:user => user, :email => params[:user_email])
+    rescue Net::SMTPAuthenticationError, Net::SMTPServerBusy, Net::SMTPSyntaxError, Net::SMTPFatalError, Net::SMTPUnknownError, Timeout::Error,
+           Errno::ECONNRESET, Errno::ECONNABORTED, Errno::EPIPE, Errno::ETIMEDOUT => e
+      Foreman::Logging.exception("Unable to deliver email", e)
+      render :json => e.message, :status => :unprocessable_entity and return
```

Comment:
I don't understand why you don't catch all exceptions, because this is called from AJAX anyway, so we do not want the default error handler to run at all. It generates huge HTML output passed in the message bubble (unreadable). Catch them all.